docs/topics/images.md

```diff
@@ -0,0 +1,372 @@
+
+Same as `url`, but always returns a full absolute URL. This requires `BASE_URL` to be set in the project settings.
+
+This is useful for images that will be re-used outside of the current site, such as social share images:
+
+```html+django
+<meta name="twitter:image" content="{{ tmp_photo.full_url }}">
+```
+
+If your site defines a custom image model using `AbstractImage`, any additional fields you add to an image (e.g. a copyright holder) are **not** included in the rendition.
+
+Therefore, if you'd added the field `author` to your AbstractImage in the above example, you'd access it using `{{ page.photo.author }}` rather than `{{ tmp_photo.author }}`.
+
+(Due to the links in the database between renditions and their parent image, you *could* access it as `{{ tmp_photo.image.author }}`, but that has reduced readability.)
+
+## The `attrs` shortcut
+
+You can also use the `attrs` property as a shorthand to output the attributes `src`, `width`, `height` and `alt` in one go:
+
+```html+django
+<img {{ tmp_photo.attrs }} class="my-custom-class" />
+```
+
+## Alternative HTML tags
+
+The `as` keyword allows alternative HTML image tags (such as `<picture>` or `<amp-img>`) to be used.
+For example, to use the `<picture>` tag:
+
+```html+django
+<picture>
+    {% image page.photo width-800 as wide_photo %}
+    <source srcset="{{ wide_photo.url }}" media="(min-width: 800px)">
+    {% image page.photo width-400 %}
+</picture>
+```
+
+And to use the `<amp-img>` tag (based on the `Mountains example <https://amp.dev/documentation/components/amp-img/#example:-specifying-a-fallback-image>`_ from the AMP docs):
+
+```html+django
+{% image image width-550 format-webp as webp_image %}
+{% image image width-550 format-jpeg as jpeg_image %}
+
+<amp-img alt="{{ image.alt }}"
+    width="{{ webp_image.width }}"
+    height="{{ webp_image.height }}"
+    src="{{ webp_image.url }}">
+    <amp-img alt="{{ image.alt }}"
+        fallback
+        width="{{ jpeg_image.width }}"
+        height="{{ jpeg_image.height }}"
+        src="{{ jpeg_image.url }}"></amp-img>
+</amp-img>
+```
+
+## Images embedded in rich text
+
+The information above relates to images defined via image-specific fields in your model. However, images can also be embedded arbitrarily in Rich Text fields by the page editor (see [](rich-text)).
+
+Images embedded in Rich Text fields can't be controlled by the template developer as easily. There are no image objects to work with, so the `{% image %}` template tag can't be used. Instead, editors can choose from one of a number of image "Formats" at the point of inserting images into their text.
+
+Wagtail comes with three pre-defined image formats, but more can be defined in Python by the developer. These formats are:
+
+### `Full width`
+Creates an image rendition using `width-800`, giving the <img> tag the CSS class `full-width`.
+
+### `Left-aligned`
+Creates an image rendition using `width-500`, giving the <img> tag the CSS class `left`.
```

Comment:
@thibaudcolas should we raise a separate issue for changing this terminology to start/end instead of left/right for 4.0. 

It would be a breaking change (or we can support both classes for a while) but our UI should probably align with everything else and avoid presumptuous about left/right alignment.